db/rdbms/migration/0002_migrate_descriptor_to_extended_descriptor.go

```diff
@@ -0,0 +1,289 @@
+		// description representing the steps, for every test. This is very ambiguous
+		// because job.JobDescriptor contains TestDescriptors itself, which however
+		// represent the higher level description of the test (including TargetManager
+		// name, TestFetcher name, etc.). So ExtendedDescriptor refers instead to
+		// StepsDescriptors and TestDescriptors field is removed from request object.
+
+		var jobDesc job.JobDescriptor
+		if err := json.Unmarshal([]byte(request.JobDescriptor), &jobDesc); err != nil {
+			return fmt.Errorf("failed to unmarshal job descriptor (%+v): %w", jobDesc, err)
+		}
+
+		var stepDescs [][]*test.StepDescriptor
+		if err := json.Unmarshal([]byte(request.TestDescriptors), &stepDescs); err != nil {
+			return fmt.Errorf("failed to unmarshal test step descriptors from request object (%+v): %w", request.TestDescriptors, err)
+		}
+
+		extendedDescriptor := job.ExtendedDescriptor{JobDescriptor: jobDesc}
+		if len(stepDescs) != len(jobDesc.TestDescriptors) {
+			return fmt.Errorf("number of tests described in JobDescriptor does not match steps stored in db")
+		}
+
+		for index, stepDesc := range stepDescs {
+			newStepsDesc := test.StepsDescriptors{}
+			// TestName is missing from the v0001 schema and can be retrieved only via
+			// TestFetcher. We need TestName in the extended_descriptor to be able to
+			// correctly build status of previous jobs. So, the only option we have is to
+			// initialize a TestFetcher and let it retrieve the test name.
+
+			for _, desc := range stepDesc {
+				newStepsDesc.Test = append(newStepsDesc.Test, *desc)
+			}
+
+			// Look up the original TestDescriptor from JobDescriptor, instantiate
+			// TestFetcher accordingly and retrieve the name of the test
+			td := jobDesc.TestDescriptors[index]
+
+			tfb, err := registry.NewTestFetcherBundle(td)
+			if err != nil {
+				return fmt.Errorf("could not instantiate test fetcher for jobID %d based on descriptor %+v: %w", request.JobID, td, err)
+			}
+
+			name, stepDescFetcher, err := tfb.TestFetcher.Fetch(tfb.FetchParameters)
+			if err != nil {
+				return fmt.Errorf("could not retrieve test description from fetcher for jobID %d: %w", request.JobID, err)
+			}
+
+			// Check that the serialization of the steps retrieved by the test fetcher matches the steps
+			// stored in the DB. If that's not the case, then, just print a warning: the underlying test
+			/// might have changed.We go ahead anyway assuming assume the test name is still relevant.
+			stepDescFetcherJSON, err := json.Marshal(stepDescFetcher)
+			if err != nil {
+				log.Warningf("steps description (`%v`) fetched by test fetcher for job %d cannot be serialized: %v", stepDescFetcher, request.JobID, err)
+			}
+
+			stepDescDBJSON, err := json.Marshal(stepDesc)
+			if err != nil {
+				log.Warningf("steps description (`%v`) fetched from db for job %d cannot be serialized: %v", stepDesc, request.JobID, err)
+			}
+
+			if string(stepDescDBJSON) != string(stepDescFetcherJSON) {
```

Comment:
**nitpicking:**

I would call it `stepDescFetched` instead of `stepDescFetcher`. Otherwise it was confusing me for few seconds while reading the code :)